### PR TITLE
Should not throw warning for unmounted TemplatesRenderer(T961348)

### PR DESCRIPTION
**Pull request by egaluza**

### CODE REVIEW

File: packages/devextreme-react/src/core/__tests__/templates-renderer.test.tsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+      expect(() => updateCallback()).not.toThrow();
+
+      component.unmount();
+      expect(ref.current).toBeNull();
+
+      expect(() => updateCallback()).not.toThrow();
+    });
+
+    it('should work deferUpdate && requestAnimationFrame', async () => {
```
Comment (LazyLahtak) [inline packages/devextreme-react/src/core/__tests__/templates-renderer.test.tsx:58]:
```suggestion
    it(`should call ${useDeferUpdate ? 'deferUpdate' : 'requestAnimationFrame'}`, async () => {
```

File: packages/devextreme-react/src/core/__tests__/templates-renderer.test.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+    beforeEach(() => {
+      (deferUpdate as jest.Mock).mockImplementation((func) => {
+        updateCallback = func;
+      });
+
+      (requestAnimationFrame as jest.Mock).mockImplementation((func) => {
+        updateCallback = func;
+      });
+    });
```
Comment (LazyLahtak) [inline packages/devextreme-react/src/core/__tests__/templates-renderer.test.tsx:35]:
```ts
const updateFunctionMock = useDeferUpdate
  ? deferUpdate as jest.Mock
  : requestAnimationFrame as jest.Mock;
  
beforeEach(() => {
  updateFunctionMock.mockImplementation((func) => {
    updateCallback = func;
  })
}
```

File: packages/devextreme-react/src/core/__tests__/templates-renderer.test.tsx (hunk 1 above)
Comment (egaluza) [reply packages/devextreme-react/src/core/__tests__/templates-renderer.test.tsx]: Replaced

File: packages/devextreme-react/src/core/__tests__/templates-renderer.test.tsx (hunk 2 above)
Comment (egaluza) [reply packages/devextreme-react/src/core/__tests__/templates-renderer.test.tsx]: Yes, I refactor this code. Thank you.

**Review by LazyLahtak — APPROVED** [review]

**Pull request merged**